File: Frontend/src/domain/models/getRecipeAllInfoDto.ts
import type { GetRecipeIngredientDto } from "./getRecipeIngredientDto";
import type { GetUnitsRecipeDto } from "./getUnitsRecipeDto";

export interface GetRecipeAllInfoDto {
  id: string;
  title: string;
  lastCooked: Date;
  ingredients: Array<GetRecipeIngredientDto>;
  recipeAmount: Array<GetUnitsRecipeDto>;
}

File: Backend/Dtos/GetRecipeAllInfoDto.cs

using Backend.Models;

namespace Backend.Dtos;

public class GetRecipeAllInfoDto
{
    public required Guid Id { get; set; }

    public required string Title { get; set; }

    public required DateTime? LastCooked { get; set; }

    public required IEnumerable<GetRecipeIngredientDto> Ingredients { get; set; }

    public required IEnumerable<GetUnitsRecipeDto> RecipeAmount { get; set; }
}